wgpu/src/lib.rs

```diff
@@ -632,6 +632,57 @@ impl Drop for ShaderModule {
     }
 }
 
+impl ShaderModule {
+    /// Get the compilation info for the shader module.
+    pub async fn compilation_info(&self) -> CompilationInfo {
+        todo!()
```

Comment:
Piping through the wgpu-core compilation message is easy enough.

But I have no idea how to deal with the Javascript promise and async Rust. Any pointers there would be hugely appreciated.